### PR TITLE
Remove SessionTicketsDisabled code after Go 1.21 update

### PR DESCRIPTION
**Pull request by jakule**
Removed lines of code in lib/auth/helpers.go and lib/utils/tls.go that disabled session tickets and managed client session cache. The behavior has changed in Go 1.21, and after the upgrade, the cache may hold old certificates that were invalidated.

This change may have a performance impact and should not be backported before full performance testing that we do during the test plan.

Related https://github.com/gravitational/teleport/pull/30201

### CODE REVIEW

**Review by codingllama — APPROVED** [review]
Curious to see what impact this may have.

LGTM as long as others are happy to have it land.

**Review by rosstimothy — APPROVED** [review]
I'm happy to merge this and see what the impacts are, but, I think we first need to define our success criteria. Will the consequences of this change be evident from our existing cases in the Test Plan?  Should we add a scenario that explicitly covers this and test it on both v13 latest and v14? Without any historical numbers how will we know if we are better or worse?

File: lib/utils/tls.go [hunk 1]
```diff
@@ -35,15 +34,13 @@ func TLSConfig(cipherSuites []uint16) *tls.Config {
...
-	// If ciphers suites were passed in, use them. Otherwise use the the
+	// If cipher suites were passed in, use them. Otherwise, use the
 	// Go defaults.
 	if len(cipherSuites) > 0 {
 		config.CipherSuites = cipherSuites
 	}
 
 	config.MinVersion = tls.VersionTLS12
-	config.SessionTicketsDisabled = false
-	config.ClientSessionCache = tls.NewLRUClientSessionCache(DefaultLRUCapacity)
```
Comment (GavinFrazar) [inline lib/utils/tls.go:46]: Note the default for `SessionTicketsDisabled` is already `false`. This change will disable client session cache, but still support session tickets for anything using this config as a server. If the intent here is to disable session tickets altogether then we should put `config.SessionTicketsDisabled = true`

Comment (GavinFrazar) [reply lib/utils/tls.go:46]: If we decide to keep session tickets enabled for servers, then I think we should also keep the client session caching and just fix the tests to not clone the cache. That can be done in the test code by calling `SetupTLSConfig` after we clone the client, or by setting it to nil or something

**Comment by zmb3** [comment]
@jakule @rosstimothy let's make a decision on this one so that it doesn't rot.

Close or merge?

**Comment by jakule** [comment]
> @jakule @rosstimothy let's make a decision on this one so that it doesn't rot.
> 
> 
> 
> Close or merge?

+1 for closing

**Pull request closed without merging**